Jenkinsfile

```diff
@@ -951,8 +987,162 @@ pipeline {
                             archiveArtifacts artifacts: 'run_test.sh/**'
                         }
                     }
-                }
-            }
+                } // stage('run_test.sh')
+                stage('Functional_Hardware_setup 1') {
+                    when {
+                        beforeAgent true
+                        allOf {
+                            expression { env.DAOS_STACK_CI_HARDWARE_SETUP != 'true' }
+                            expression {
+                              ! commitPragma(pragma: 'Skip-func-hw-wetup').contains('true')
+                            }
+                        }
+                    }
+                    agent {
+                        label 'stage_nvme4'
+                    }
+                    steps {
+                        provisionNodes NODELIST: env.NODELIST,
+                                       node_count: 4,
+                                       profile: 'daos_ci',
+                                       inst_repos: el7_daos_repos,
+                                       inst_rpms: 'daos-' + daos_packages_version +
+                                                  ' daos-client-' + daos_packages_version +
+                                                  ' cart-' + env.CART_COMMIT + ' ' +
+                                                  el7_functional_rpms
+                        runTest stashes: [ 'CentOS-install', 'CentOS-build-vars' ],
+                                script: '''tnodes=$(echo $NODELIST | cut -d ',' -f 1-4)
+                                           # set DAOS_TARGET_OVERSUBSCRIBE env here
+                                           export DAOS_TARGET_OVERSUBSCRIBE=1
+                                           . ./.build_vars.sh
+                                           DAOS_BASE=${SL_PREFIX%/install*}
+                                           clush -B -S -o '-i ci_key' -l jenkins -w ${tnodes} \
+                                               "sudo mkdir -p $DAOS_BASE;sudo mount -t nfs $HOSTNAME:$PWD $DAOS_BASE"
+
+                                           clush -B -S -o '-i ci_key' -l jenkins -w ${tnodes} \
+                                             "if [ -e /dev/pmem0 ]; then sudo ${DAOS_BASE}/install/bin/daos_server storage prepare --scm-only --reset; fi"''',
```

Comment:
These can be combined into a single command, and you should put a `umount` in an `EXIT` `trap` so that it's always done no matter how the scriptlet exits:

```suggestion
                                           clush -B -S -o '-i ci_key' -l jenkins -w ${tnodes} \
                                               "set -e
                                                sudo mkdir -p $DAOS_BASE
                                                sudo mount -t nfs $HOSTNAME:$PWD $DAOS_BASE
                                                trap 'sudo umount $DAOS_BASE' EXIT
                                                if [ -e /dev/pmem0 ]; then
                                                    sudo ${DAOS_BASE}/install/bin/daos_server storage prepare --scm-only --reset
                                                fi"''',
```

From the Jenkins run, it looks like you need some kind of _force_ option for `daos_server storage prepare` as it's prompting for a confirmation:

```
[2020-02-27T00:24:16.605Z] Resetting locally-attached SCM...
[2020-02-27T00:24:16.605Z] Memory allocation goals for SCM will be changed and namespaces modified, this will be a destructive operation. Please ensure namespaces are unmounted and locally attached SCM & NVMe devices are not in use. Please be patient as it may take several minutes and subsequent reboot maybe required.
[2020-02-27T00:24:16.605Z] Are you sure you want to continue? (yes/no)
[2020-02-27T00:24:16.605Z] ERROR: Error reading input: EOF
[2020-02-27T00:24:16.605Z] ERROR: scan error(s):
[2020-02-27T00:24:16.605Z]   consent not given
```

I see that the functional test framework is using `-n` and `-f` arguments on one of it's `storage prepare` commands.  My bet is `-f` is to force/skip confirmation.